dashboard/lib/programming_expression_autocomplete.rb

```diff
@@ -1,14 +1,19 @@
 class ProgrammingExpressionAutocomplete < AutocompleteHelper
-  def self.get_search_matches(query, limit, programming_environment)
+  def self.get_search_matches(page, query, programming_environment)
     query = format_query(query)
     return [] if query.length < MIN_WORD_LENGTH
 
-    limit = format_limit(limit)
-    rows = ProgrammingExpression.limit(limit)
+    rows = ProgrammingExpression.all
     rows = rows.where(programming_environment: programming_environment) if programming_environment
 
     rows = rows.
         where("MATCH(name,category) AGAINST(? in BOOLEAN MODE)", query)
-    return rows.map(&:summarize_for_lesson_edit)
+
+    rows = rows.limit(100)
+    total_rows = rows.length
+    page_number = (total_rows / 10.0).ceil
+    rows = rows.page(page).per(10)
```

Comment:
It's not immediately clear to me how performant this is; do you have a sense of that? I'd love to see a comment explaining

Reply:
haha, oh yeah in that case definitely agreed